src/main/java/com/intershop/version/semantic/VersionResolver.java

```diff
@@ -0,0 +1,245 @@
+package com.intershop.version.semantic;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+class VersionResolver implements SemanticVersionResolver
```

Comment:
The naming seems to be a little weird: the interface name sounds more specific than its implementation. Maybe rename it.